plugins/qeta-backend/src/service/router.ts

```diff
@@ -864,6 +868,192 @@ export async function createRouter({
+          const userRef = `user:default/${loggedUser.toString()}`;
+          const loggedUserUpvotedQuestions =
+            await database.getMostUpvotedQuestions({
+              author: userRef,
+              options,
+            });
+
+          if (loggedUserUpvotedQuestions) {
+            rankingResponse.loggedUser!.author =
+              loggedUserUpvotedQuestions.length > 0
+                ? loggedUserUpvotedQuestions[0].author
+                : userRef;
+
+            rankingResponse.loggedUser!.total =
+              loggedUserUpvotedQuestions.length > 0
+                ? loggedUserUpvotedQuestions[0].total
+                : 0;
+
+            rankingResponse.loggedUser!.position = `${mostUpvotedQuestions.length}+`;
+          }
+        } else {
+          rankingResponse.loggedUser = findLoggerUserInData;
+        }
+      }
+
+      return response.status(200).json(rankingResponse);
+    },
+  );
+
+  // GET /statistics/answers/top-upvoted-users?period=x&limit=x&loggedUser=x
+  router.get(
+    '/statistics/answers/top-upvoted-users',
+    async (request, response) => {
+      const { period, limit, loggedUser } = request.query;
+
+      const options: StatisticsOptions = {
+        period: period && stringDateTime(period?.toString()),
+        limit: Number(limit),
+        loggedUser: loggedUser?.toString(),
+      };
+
+      const mostUpvotedAnswers: Statistic[] =
+        await database.getMostUpvotedAnswers({
+          options,
+        });
+
+      const rankingResponse = {
+        ranking: mostUpvotedAnswers,
+        loggedUser: {},
+      } as StatisticResponse;
+
+      if (mostUpvotedAnswers.length === 0) {
+        return response.status(204).json(rankingResponse);
+      }
+
+      if (loggedUser) {
+        const findLoggerUserInData = mostUpvotedAnswers.find(userStats => {
+          return userStats.author?.includes(loggedUser.toString());
+        });
+
+        if (!findLoggerUserInData) {
+          const userRef = `user:default/${loggedUser.toString()}`;
+          const loggedUserUpvotedAnswers = await database.getMostUpvotedAnswers(
+            {
+              author: userRef,
+              options,
+            },
+          );
+
+          if (loggedUserUpvotedAnswers) {
+            rankingResponse.loggedUser!.author =
+              loggedUserUpvotedAnswers.length > 0
+                ? loggedUserUpvotedAnswers[0].author
+                : userRef;
+
+            rankingResponse.loggedUser!.total =
+              loggedUserUpvotedAnswers.length > 0
+                ? loggedUserUpvotedAnswers[0].total
+                : 0;
+
+            rankingResponse.loggedUser!.position = `${mostUpvotedAnswers.length}+`;
+          }
+        } else {
+          rankingResponse.loggedUser = findLoggerUserInData;
+        }
+      }
+
+      return response.status(200).json(rankingResponse);
+    },
+  );
+
+  // GET /statistics/answers/top-correct-upvoted-users?period=x&limit=x&loggedUser=x
+  router.get(
```

Comment:
Not related to this change, but the router is soon getting out of hand and should be split into separate files

Reply:
I totally agree 😅